sklearn/base.py

```diff
@@ -458,6 +458,14 @@ def fit_predict(self, X, y=None):
         self.fit(X)
         return self.labels_
 
+    @property
+    def n_features_out_(self):
+        if not hasattr(self, 'transform'):
+            raise AttributeError("{} doesn't have n_features_out_"
```

Comment:
When I was working on `feature_names_out_`, I realized you could think of the `feature_names_out_` of classifier xxx, as `xxx0` to `xxx{n-1}` for a multiclass classification for instance. That would also make sense in the context of a stacking estimator. What are the `feature_names_in_` of a stacking estimator if we're talking about interpreting the model? Should it not be `{classifier_{i}_{c} | for i in classifiers and c in classes}`, for instance?

Although you could implement a stacking estimator also with a meta-estimator which defines `transform` and returns the output of the estimator, do the union of those, then have a model on top of that. Then the problem's solved.

Reply:
Sorry, I'm not sure what your point is. This is for the ClusterMixin.
And I thought we only wanted to define feature names out for transformers.
Stacking should either be a transformer or a meta-estimator.
Also check what this PR does for ``VotingClassifier`` btw. But I'm really not sure I follow.